components/catalogs/back/tests/Integration/Infrastructure/Controller/Internal/GetCategoriesActionTest.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Catalogs\Test\Integration\Infrastructure\Controller\Internal;
+
+use Akeneo\Catalogs\Test\Integration\IntegrationTestCase;
+use PHPUnit\Framework\Assert;
+
+/**
+ * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class GetCategoriesActionTest extends IntegrationTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->purgeDataAndLoadMinimalCatalog();
+    }
+
+    public function testItGetsCategories(): void
+    {
+        $client = $this->getAuthenticatedInternalApiClient();
+
+        $tshirtCategory = $this->createCategory(['code' => 'tshirt', 'labels' => ['en_US' => 'T-shirt']]);
+        $shoesCategory = $this->createCategory(['code' => 'shoes', 'labels' => ['en_US' => 'Shoes']]);
+
+        $client->request(
+            'GET',
+            '/rest/catalogs/categories',
+            ['codes' => 'tshirt,shoes'],
+            [],
+            [
+                'HTTP_X-Requested-With' => 'XMLHttpRequest',
+            ],
+        );
+
+        $response = $client->getResponse();
+        Assert::assertEquals(200, $response->getStatusCode());
+
+        $categories = \json_decode($response->getContent(), true);
+
+        $expectedTshirtCategory = [
+            'id' => $tshirtCategory->getId(),
```

Comment:
```suggestion
            'id' => $tshirtCategory->getId(),
```

Feels like a red flag. Same question as above, do we need to expose the id ?
If the only reason is because we need it to call `/rest/catalogs/categories/{categoryId}/children`, then, we could also change this other route to `/rest/catalogs/categories/{code}/children`.

Reply:
You are right ! CategoryRepository can only fetch children through parent Id
But yes I could fetch the parent category first to get Id in the children query beforehand (assuming Category Service API would have a similar interface)
Will change!